### PR TITLE
feat(neuron-ui): add history and transaction detail view

### PR DESCRIPTION
**Pull request by Keith-CY**
Loading style needs to be determined.

### CODE REVIEW

File: packages/neuron-ui/.env [hunk 1]
```diff
@@ -1 +1,2 @@
 REACT_APP_NETWORKS=[{"name":"Testnet","remote":"http://localhost:8114"},{"name":"Localhost","remote":"http://localhost:8114"}]
+REACT_EXPLORER=http://localhost:3000
```
Comment (ashchan) [inline packages/neuron-ui/.env]: Please move both `REACT_APP_NETWORKS` and `REACT_EXPLORER` to in-code constants (and update names as well). For the foreseen period there's no need to make it so flexible to be configurable via ENV.

File: packages/neuron-ui/src/components/History/index.tsx [hunk 2]
```diff
@@ -1,63 +1,172 @@
...
+          <div>{i18n.t(`history.${TransactionType[tx.type as TransactionType]}`.toLowerCase())}</div>
+          <div>{time}</div>
+        </>
+      ),
+      key: tx.hash,
+      date,
+    }
+  })
+
+const groupHistory = (items: Transaction[]) => {
```
Comment (ashchan) [inline packages/neuron-ui/src/components/History/index.tsx]: How about grouping transactions first, then constructing the list (`JSX.Element`s).

The former is how we organize the data, while the later is for UI display.

File: packages/neuron-ui/src/containers/MainContent/actionCreators/networks.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  saveNetworks: (idx: number, networks: Network[], editorNetwork: Network, navTo: (path: string) => void) => {
+    if (!editorNetwork.name) {
+      return {
+        type: MainActions.ErrorMessage,
+        payload: {
+          networks: Message.NameIsRequired,
+        },
+      }
+    }
+    if (editorNetwork.name.length > 28) {
```
Comment (ashchan) [inline packages/neuron-ui/src/containers/MainContent/actionCreators/networks.ts]: Please make `28` a constant.

File: packages/neuron-ui/.env (hunk 1 above)
Comment (Keith-CY) [reply packages/neuron-ui/.env]: REACT_APP_NETWORKS will be removed by another PR which move network management to neuron-wallet.

REACT_EXPLORER will be removed soon.

**Comment by Keith-CY** [comment]
> There're many changes that are not related to history/transaction feature but are arch and other refactorings/improvements. Next time please separate them into smaller PRs.

My fault, I should do it earlier.

**Comment by ashchan** [comment]
Not a problem, cheers on the refactoring. 👍

File: packages/neuron-ui/src/components/History/index.tsx (hunk 2 above)
Comment (Keith-CY) [reply packages/neuron-ui/src/components/History/index.tsx]: I've tried before, it requires one more step to format data. But it's indeed more clearly. I'll do it now.

File: packages/neuron-ui/src/containers/MainContent/actionCreators/networks.ts (hunk 3 above)
Comment (Keith-CY) [reply packages/neuron-ui/src/containers/MainContent/actionCreators/networks.ts]: updated in https://github.com/nervosnetwork/neuron/pull/166/commits/a3a5908cfa0a67cee9ff7c6f5c79906c16e9e5a7

File: packages/neuron-ui/.env (hunk 1 above)
Comment (Keith-CY) [reply packages/neuron-ui/.env]: updated in https://github.com/nervosnetwork/neuron/pull/166/commits/ba8fae7ebe7c88d03cad0879d52c0b7071b012f3

File: packages/neuron-ui/src/components/History/index.tsx (hunk 2 above)
Comment (Keith-CY) [reply packages/neuron-ui/src/components/History/index.tsx]: updated in https://github.com/nervosnetwork/neuron/commit/051ad900754fc53de19ba414d575547058bd125e

**Pull request merged**